src/Storages/VirtualColumnUtils.h

```diff
@@ -74,9 +81,11 @@ struct VirtualsForFileLikeStorage
     std::optional<size_t> size { std::nullopt };
     const String * filename { nullptr };
     std::optional<Poco::Timestamp> last_modified { std::nullopt };
-
+    const String & hive_partitioning_path = "";
```

Comment:
Actually, we can avoid adding extra field for hive partitioning. We calready have path in `path` field, so we can use it and just propogate context to `addRequestedFileLikeStorageVirtualsToChunk` and check `use_hive_partitioning` there